src/main/scala/abalone/Pos.scala

```diff
@@ -190,7 +190,43 @@ case class Pos private (index: Int) extends AnyVal {
   def right: Option[Pos]     = Pos.at(file.index + 1, rank.index)
   def downRight: Option[Pos] = Pos.at(file.index, rank.index - 1)
   def upRight: Option[Pos]   = Pos.at(file.index + 1, rank.index + 1)
-  def neighbours: List[Option[Pos]] = List(left, downLeft, upLeft, right, downRight, upRight)
+
+  def neighbours: List[Option[Pos]] = List(left, upLeft, upRight, right, downRight, downLeft)
+  def sideMovesDirsFromDir(dir: Option[String]): (Option[Pos], Option[Pos]) = dir match {
+    case Some("left")       => (this.downLeft, this.upLeft)
+    case Some("upLeft")     => (this.left, this.upRight)
+    case Some("upRight")    => (this.upLeft, this.right)
+    case Some("right")      => (this.upRight, this.downRight)
+    case Some("downRight")  => (this.right, this.downLeft)
+    case Some("downLeft")   => (this.downRight, this.left)
+    case _                  => (None, None)
+  }
+
+  // NOTE - *neighbourhood
+  // these below only work for neighbour pos but that's probably fine as in Abalone we only move to (potentially extended) neighbourhood
+  def dir(dir: Option[String]): Option[Pos]  = dir match {
```

Comment:
still using String for direction...

Reply:
So did you work out a way to reuse `Direction` here? I see `Directions` used in other places in the code